subsys/bluetooth/shell/bt.c

```diff
@@ -84,6 +87,13 @@ enum {
 	SHELL_ADV_OPT_NUM,
 };
 
+#if defined(CONFIG_BT_OBSERVER)
+typedef struct scan_data {
```

Comment:
We already have a `#if defined(CONFIG_BT_OBSERVER)` where we defined e.g. `struct bt_scan_filter`. Suggest to use the same guard :)